### PR TITLE
Persisted and protected IDevID and IAK test

### LINKED ISSUE #587: Add a test to cover IDevID/IAK usage with passwords
The goal is to cover the changes introduced by https://github.com/keylime/rust-keylime/pull/785

@Isaac-Matthews It would be great if you could contribute a test case

### PR DESCRIPTION
**Pull request by Isaac-Matthews**
Fixes #587 
This is a test for IDevIDs and IAKs that are persisted in the TPM and are accessed via handle rather than regenerated. These keys may also be protected by password, this is also tested, as is the change to enable hex passwords for the TPM.

Relevant PRs:
[agent: enable persisted and protected IDevIDs](https://github.com/keylime/rust-keylime/pull/785), [agent: enable hex passwords for the TPM](https://github.com/keylime/rust-keylime/pull/769), [server: add config options](https://github.com/keylime/keylime/commit/7b3ea56aae40dc04ae4f28f0147486f90ff547fa)

### CODE REVIEW

**Comment by kkaarreell** [comment]
Please rebase, that should help to get rid of most of test failures. Thank you.

File: functional/iak-idevid-persisted-and-protected/test.sh
Comment (kkaarreell) [inline functional/iak-idevid-persisted-and-protected/test.sh:52]: Usually, we use [openssl-certgen beakerlib library](https://pagure.io/beakerlib-libraries/blob/master/f/openssl/certgen/lib.sh) to generate certificates, e.g. https://github.com/RedHat-SP-Security/keylime-tests/blob/7cad5f9dbb5bb1fb365b2a418ec75feba4f51d51/functional/basic-attestation-with-custom-certificates/test.sh (plus a require item in main.fmf)

Are there "ordinary" certificates or is there anything "special" about them? I am not strongly against doing it this way, even though I believe the other approach would be shorter..  But in cases these certs have any special property it would be good to mention it in a comment.

**Review by kkaarreell — APPROVED** [review]

File: functional/iak-idevid-persisted-and-protected/test.sh
Comment (kkaarreell) [reply functional/iak-idevid-persisted-and-protected/test.sh:52]: Hi @ansasaki , could you please review it as well?

Comment (ansasaki) [reply functional/iak-idevid-persisted-and-protected/test.sh:52]: Yes, I'll review

Comment (ansasaki) [reply functional/iak-idevid-persisted-and-protected/test.sh:52]: Sorry for taking long to get to review this.

I agree with what Karel wrote, I believe it would be quite simple to use the `certgen/certgen` beaker library to generate the certificates including the required SAN without having to carry the CA configuration files.

In my opinion it would clarify what is important to have in the certificate.

If you disagree or cannot make the changes, I think it is acceptable to keep the configuration files for now.

The only change I would like to request is to provide a better commit message.

Comment (Isaac-Matthews) [reply functional/iak-idevid-persisted-and-protected/test.sh:52]: No problem, thanks for the review. I will change the commit message.

For the certs - I don't disagree. However can I suggest we merge this first, and then when I have some time I will look into changing over to using certgen. This would also mean that the change can affect both this test and the other IDevID test as swapping the cert generation mechanism should probably be part of the same commit (the other test currently uses the same process that is used here).

**Comment by kkaarreell** [comment]
Hi @Isaac-Matthews , it seems that all the related changes are already merged upstream. Can we merge this test PR as well?
FYI, in order to run this test in keylime CI we need to add it to the packit-ci.fmf plan in the upstream project.

**Comment by Isaac-Matthews** [comment]
> Hi @Isaac-Matthews , it seems that all the related changes are already merged upstream. Can we merge this test PR as well? FYI, in order to run this test in keylime CI we need to add it to the packit-ci.fmf plan in the upstream project.

@kkaarreell Yes this is ready to be merged, I was just waiting for reviews. So please merge whenever.
For the plans I already have draft prs [1](https://github.com/keylime/rust-keylime/pull/814) [2](https://github.com/keylime/keylime/pull/1589) to enable the test. If any other changes are needed let me know.

**Pull request merged**
